lib/arjdbc/postgresql/adapter.rb

```diff
@@ -866,6 +852,11 @@ def column_definitions(table_name)
            AND a.attnum > 0 AND NOT a.attisdropped
          ORDER BY a.attnum
       end_sql
+
+      # Force the notnull attribute to a boolean
+      rows.each do |row|
+        row[3] = row[3] == 't' if row[3].is_a?(String)
```

Comment:
This previously happened in the #columns method but doing it here allows us to let the ActiveRecord adapter handle the bulk of the work.